scripts/build.py

```diff
@@ -382,8 +381,9 @@ def hash_should_be_inserted(filepath):
 
 
 def copy_files_source_to_target(source, target, file_hashes):
-    """Copies all files in source directory to target directory and renames
-    them to include hash in their name.
+    """Copies all files in source directory to target directory, renames
+    them to include hash in their name and remove files that are not in the hash
```

Comment:
I will incorporate this delete login into build_files. Seems like it is more natural there?

Reply:
EDIT: On 2nd thought, since deleting a file (or renaming files thus deleting the original ) counts a file changes, I will handle this event with newly added `get_recently_changed_filenames` and `rebuid_new_files`

Reply:
For clarification purpose, these deletion logic works like this:

DEV files --> Add MD5 Hash to hash dict and Minified into staging dir(`/backend_prod_files`) -->Copied into final directory(`/build`) and append hash to filename

A file(let's say base.html) is deleted(or renamed) in /DEV --> The existing file in staging directory is untouched --> File, previously staged and built, gets built and copied into final directory which is not supposed to happen.

This was a side-effect of not nuking the staging dir (/backend_prod_files)  every time a prod build is trigger.  Before, every single files would be deleted and completely rebuilt. Now we only update newly edited files and left the deleted ones behind.

Reply:
So I have been doing some brainstorming: There are potential solution I can think of right now for integrating the "deletion" or clean up logic here:
1) Create an independent walk of staging dir, whichever files not found in hash dict will be deleted.
2) Rename `copy_files_source_to_target` to `process_staging_dir`. `process_staging_dir` will now handle 2 main logic:
   a) Create the hashed filename (`base.240933e7564bd72a4dde42ee23260c5f.html`) and copy the minified `base.html` over with the hashed name.
   b) If a file does not have a matching hash entry, delete that file. Continue onto next file.

What do you think, @seanlip?

Reply:
Both solutions seem reasonable. (1) is probably better since it separates out the logic a bit more (and if we were ever to unit test this, it would make it easier to do so).

Reply:
Alright, I will rolling with (2). Looks like the additional walk will takes another 30 seconds, but I think it is just cleaner to separate version control logic.